src/he/flow.c

```diff
@@ -60,6 +61,15 @@ he_return_code_t he_conn_inside_packet_received(he_conn_t *conn, uint8_t *packet
     return HE_ERR_UNSUPPORTED_PACKET_TYPE;
   }
 
+  // Clamp the MSS if PMTU has been fixed
+  if(conn->pmtud_state == HE_PMTUD_STATE_SEARCH_COMPLETE) {
+    uint16_t effective_pmtu = he_conn_get_effective_pmtu(conn);
+    he_return_code_t ret = he_clamp_mss(packet, length, effective_pmtu - HE_MSS_OVERHEAD);
+    if(ret != HE_SUCCESS) {
+      return ret;
+    }
+  }
```

Comment:
Shall we add a check to only do this on client-side?

Reply:
Update: it's correct that we should do mss clamping on _both_ client and server sides, because the mss value can be different on each of the direction. However, on the client, it should set mss value on the TCP SYN packet, but on the server side, I think it should set the mss value on the TCP ACK-SYN packet.